src/main/java/com/khutircraftubackend/auth/exception/UserExceptionHandler.java

```diff
@@ -1,48 +1,72 @@
-     *
-     * @param e the e
-     * @return the response entity
-     */
-    @ExceptionHandler(UnauthorizedException.class)
-    public ResponseEntity<String> handleUnauthorizedException(UnauthorizedException e) {
-        return new ResponseEntity<>(e.getMessage(), HttpStatus.UNAUTHORIZED);
+    @ExceptionHandler(BadCredentialsException.class)
+    public ResponseEntity<Map<String, Object>> handleBadCredentialsException(
+            BadCredentialsException ex, HttpServletRequest request) {
+
+        Map<String, Object> errors = setErrors(403, "Не правільні облікові дані.",
+                ex.getMessage(), request.getRequestURI());
+
+        return ResponseEntity.status(403).body(errors);
     }
 
-    /**
-     * Handles UserNotFoundException and returns a response with status 404 (Not Found).
-     *
-     * @param e the exception to handle
-     * @return the response entity with the exception message and status 404
-     */
-    @ExceptionHandler(UserNotFoundException.class)
-    public ResponseEntity<String> handleUserNotFoundException(UserNotFoundException e) {
-        return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Користувач не знайдений: " + e.getMessage());
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<Map<String, Object>> handleValidationExceptions(
+            MethodArgumentNotValidException ex, HttpServletRequest request) {
+        Map<String, List<String>> errorsMessage = ex.getBindingResult().getFieldErrors().stream()
+                .collect(Collectors.groupingBy(
+                        FieldError::getField,
+                        LinkedHashMap::new,
+                        Collectors.mapping(FieldError::getDefaultMessage, Collectors.toList())
+                ));
+        Map<String, Object> errors = setErrors(ex.getStatusCode().value(),
+                HttpStatus.valueOf(ex.getStatusCode().value()).getReasonPhrase(),
+                errorsMessage,
+                request.getRequestURI());
+
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
     }
 
-    /**
-     * Handle bad credentials exception response entity.
-     *
-     * @param e the e
-     * @return the response entity
-     */
-    @ExceptionHandler(BadCredentialsException.class)
-    public ResponseEntity<String> handleBadCredentialsException(BadCredentialsException e) {
-        return new ResponseEntity<>("Неправильні облікові дані", HttpStatus.UNAUTHORIZED);
+    @ExceptionHandler(ResponseStatusException.class)
+    public ResponseEntity<Map<String, Object>> getErrorAttributes(
+            ResponseStatusException ex, HttpServletRequest request) {
+        Map<String, Object> errors = setErrors(ex.getStatusCode().value(),
+                HttpStatus.valueOf(ex.getStatusCode().value()).getReasonPhrase(),
+                ex.getReason(),
+                request.getRequestURI());
+
+        return ResponseEntity.status(ex.getStatusCode()).body(errors);
     }
```

Comment:
no need to handle `ResponseStatusException` 
it'll be handled by Spring Boot internally. 
see examples here: https://github.com/Hutiir-Craft-Food/learn-exception-handling